src/datatrove/pipeline/tokens/tokenizer.py

```diff
@@ -222,6 +250,8 @@ def __init__(
         self.local_working_dir = get_datafolder(local_working_dir) if local_working_dir else None
         if self.local_working_dir and not isinstance(self.local_working_dir.fs, LocalFileSystem):
             raise ValueError("local_working_dir must be a local path")
+        if self.local_working_dir is None:
```

Comment:
The check should be:
If local_working_dir is None AND shuffle is True AND output_dir is not a local directory

(view line above for this last one)